spec.html

```diff
@@ -30110,19 +30110,59 @@ <h1>Function.prototype.toString ( )</h1>
           1. Let _func_ be the *this* value.
           1. If Type(_func_) is Object and _func_ has a [[SourceText]] internal slot and _func_.[[SourceText]] is a sequence of Unicode code points and HostHasSourceTextAvailable(_func_) is *true*, then
             1. Return CodePointsToString(_func_.[[SourceText]]).
-          1. If _func_ is a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ has an [[InitialName]] internal slot and _func_.[[InitialName]] is a String, the portion of the returned String that would be matched by |NativeFunctionAccessor?| |PropertyName| must be the value of _func_.[[InitialName]].
-          1. If Type(_func_) is Object and IsCallable(_func_) is *true*, return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|.
-          1. Throw a *TypeError* exception.
+          1. Else if _func_ is a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, then
+            1. Let _sourceText_ be an implementation-defined sequence of Unicode code points representing _func_.
+          1. Else if Type(_func_) is Object and IsCallable(_func_) is *true*, then
+            1. Let _sourceText_ be an implementation-defined sequence of Unicode code points representing _func_.
+          1. Else,
+            1. Throw a *TypeError* exception.
+          1. Assert: ParseText(_sourceText_, |NativeFunction|) is a Parse Node.
```

Comment:
This assert may not hold. You have to say that `_sourceText_` has the syntax of `|NativeFunction|` somewhere, otherwise it's just any arbitrary text.

Reply:
Fixed.